test/apps/glsl-debugger/layer-debugger.js

```diff
@@ -0,0 +1,67 @@
+import {_DebugContext as DebugContext, COLOR_MODE} from '@luma.gl/debug';
+
+const DEFAULT_OPTIONS = {
+  enabled: false,
+  colorMode: COLOR_MODE.NONE
+};
+
+export default class LayerDebugger {
+  constructor(deck, opts) {
+    this.deck = deck;
+    this.opts = Object.assign({}, DEFAULT_OPTIONS, opts);
+
+    deck.setProps({
+      onBeforeRender: this._onRender.bind(this)
+    });
+
+    this.debugContext = new DebugContext(deck.canvas);
+  }
+
+  setOptions(opts) {
+    Object.assign(this.opts, opts);
+
+    // Hack: trigger rerender
+    this.deck.setProps({
+      layers: [...this.deck.props.layers]
+    });
+  }
+
+  _onRender() {
+    const {deck} = this;
+
+    this.debugContext.clear({
+      strokeStyle: '#000',
+      fillStyle: '#000'
+    });
+
+    const {layerManager} = deck;
+    layerManager.layers = layerManager.layers.map(this._inject, this);
+  }
+
+  _inject(layer) {
+    if (!layer.__debugger) {
+      const originalDraw = layer.draw;
+
+      layer = Object.create(layer);
+
+      layer.draw = (...args) => {
+        originalDraw.call(layer, ...args);
+        this._drawLayer(layer);
+      };
+      layer.__debugger = this;
```

Comment:
How can this work? I.e. adding a `__debugger` field here. We `Object.seal` the layer in the constructor. Is `layer` not an actual `Layer` instance?

Reply:
Correct, `layer` is just a plain object with reference to the methods of the original `Layer` instance.